### PR TITLE
Bugfix/update options link ref

### LINKED ISSUE #390: (Link|Reference)Input: Options are not refreshed when a new component is added while the ComponentDetailsPanel is open
**Describe the bug**

When I select a component with a `Link`/`Reference` attribute, then add a component that can be targeted by the `Link`/`Reference` attribute, the `options` of the attribute are not refreshed while the `ComponentDetailsPanel` is open.

I expect it to be automatically refreshed.

**Note:** This bug is similar to https://github.com/ditrit/leto-modelizer-plugin-core/issues/218.

**To Reproduce**

https://github.com/ditrit/leto-modelizer/assets/20184007/d54bf24c-cdbe-4d05-9bec-c28cdd1ca642

https://github.com/ditrit/leto-modelizer/assets/20184007/96dfb65c-3ee8-4741-bcd0-0105564ae294

**Application**

- `leto-modelizer` commit hash: `bf29a9d802b57b50ee00c8000f13561f884b00bc`.
- `leto-modelizer-plugin-core` version: `#0.19.0`.

### PR DESCRIPTION
**Pull request by k-y0u**
Fixes #390

### CODE REVIEW

File: cypress/e2e/Bugfix/Issue390.feature [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    And  I click on '[data-cy="create-model-form"] [data-cy="submit-button"]'
+    And  I wait 1 second
+    Then I expect 'positive' toast to appear with text 'Model has been created 🥳!'
+    And  I expect current url is 'projectName/modelizer/draw\?plugin=terrator-plugin&path=infraFirst'
+
+    # Create one link option & one ref option
+    When I click on '[data-cy="component-definitions-item_terrator-plugin"]'
+    And  I click on '[data-cy="component-definition_aws_route53_zone"]'
+    And  I click on '[data-cy="component-definition_aws_route53_record"]'
+    Then I expect '[id^="aws_route53_zone"]' appear 1 times on screen
```
Comment (justinehell) [inline cypress/e2e/Bugfix/Issue390.feature:29]: appear 1 time
-> fix the other lines on this file

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/components/inputs/LinkInput.vue [hunk 2]
```diff
@@ -43,17 +51,20 @@ const iconName = ref(plugin.value.data.definitions.components.find(
...
-watch(() => linkInput.value, () => {
-  if (linkInput.value) {
-    linkInput.value.validate();
+/**
+ * Update input options.
+ * @param {object} eventManager - Object containing event and plugin.
+ * @param {object} eventManager.event - The triggered event.
+ */
+function updateOptions({ event }) {
+  if (event.type === 'Drawer' && event.status === 'success') {
```
Comment (Zorin95670) [inline src/components/inputs/LinkInput.vue:62]:
```js
if (event.type === 'Drawer' && event.status === 'success' && event.action !== 'move' && event.action !== 'resize') {
```

File: src/components/inputs/ReferenceInput.vue [hunk 3]
```diff
@@ -37,23 +45,25 @@ const referenceInput = ref(null);
...
-watch(() => referenceInput.value, () => {
-  if (referenceInput.value) {
-    referenceInput.value.validate();
+/**
+ * Update input options.
+ * @param {object} eventManager - Object containing event and plugin.
+ * @param {object} eventManager.event - The triggered event.
+ */
+function updateOptions({ event }) {
+  if (event.type === 'Drawer' && event.status === 'success') {
```
Comment (Zorin95670) [inline src/components/inputs/ReferenceInput.vue:61]:
```js
if (event.type === 'Drawer' && event.status === 'success' && event.action !== 'move' && event.action !== 'resize') {
```

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: cypress/e2e/Bugfix/Issue390.feature (hunk 1 above)
Comment (k-y0u) [reply cypress/e2e/Bugfix/Issue390.feature]: done

File: src/components/inputs/ReferenceInput.vue (hunk 3 above)
Comment (k-y0u) [reply src/components/inputs/ReferenceInput.vue]: done

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**